py/dml/c_backend.py

```diff
@@ -1745,13 +1745,14 @@ def generate_init(device, initcode, outprefix):
     out('}\n\n', preindent = -1)
     splitting_point()
 
-def generate_extern_trampoline(exported_name, func):
-    params = (func.cparams if func.independent else
-              [("_obj", TPtr(TNamed("conf_object_t")))] + func.cparams[1:])
-    out("extern %s\n" % (func.rettype.declaration(
-                         "%s(%s)" % (exported_name,
-                                     ", ".join(t.declaration(n)
-                                               for (n, t) in params)))))
+def generate_static_trampoline(func):
+    # static trampolines never need to be generated for independent methods
+    assert not func.independent
+    params = [("_obj", TPtr(TNamed("conf_object_t")))] + func.cparams[1:]
+    params_string = ('void' if not params
+                     else ", ".join(t.declaration(n) for (n, t) in params))
+    start_function_definition(func.rettype.declaration(
+        "%s(%s)" % ("_trampoline" + func.get_cname(), params_string)))
```

Comment:
Cosmetics: I suppose this turns into something like `_trampolineDML_M_foo`? So I suggest `_` after trampoline.

Reply:
No, it doesn't, cnames for methods already have a `_` prefix, so it turns into `_trampoline_DML_M_foo`.